tests/testthat/test-hjust-text-anchor.R

```diff
@@ -0,0 +1,115 @@
+  i <- 1
+  while (gap > tol && i <= nmax) {
+    xy <- rbind(xy, newxy[i, ])
+    newxy <- rbind(newxy, xy[i + 1, ] - gamma * attr(grad(xy[i + 1, 1], xy[i + 1, 2]), 'gradient'))
+    z <- c(z, FUN(newxy[i + 1, 1], newxy[i + 1, 2]))
+    gap <- abs(z[i + 1] - FUN(xy[i + 1, 1], xy[i + 1, 2]))
+    i <- i + 1
+    if (i > nmax) warning('Maximum number of iterations reached!')
+  }
+  objective <- data.frame(iteration = 1:i, x = xy[, 1], y = xy[, 2], z = z)
+  invisible(list(contour = contour, objective = objective))
+}
+
+dat <- grad.desc()
+contour <- dat$contour
+objective <- dat$objective
+objective <- ldply(objective$iteration, function(i) {
+  df <- subset(objective, iteration <= i)
+  cbind(df, iteration2 = i)
+})
+objective2 <- subset(objective, iteration == iteration2)
+
+grad.desc.viz <- function(hjust) {
+  if (!hjust %in% c(0, 0.5, 1)) {
+    warning("unsupported value of hjust was provided.")
+  }
+  objective2$hjust <- hjust
+  
+  contour.plot <- ggplot() + 
+    geom_contour(data = contour, aes(x = x, y = y, z = z, colour = ..level..), size = .5) + 
+    scale_colour_continuous(name = "z value") + 
+    geom_path(data = objective, aes(x = x, y = y, showSelected = iteration2), 
+              colour = "red", size = 1) + 
+    geom_point(data = objective, aes(x = x, y = y, showSelected = iteration2), colour = "green", 
+               size = 2) + 
+    geom_text(data = objective2, aes(x = x, y = y - 0.2, showSelected = iteration2, label = round(z, 2))) + 
+    scale_x_continuous(expand = c(0, 0)) + 
+    scale_y_continuous(expand = c(0, 0)) + 
+    ggtitle("contour of function value") + 
+    theme_animint(width = 600, height = 600)
+  
+  objective.plot <- ggplot() +
+    geom_line(data = objective2, aes(x = iteration, y = z), colour = "red") + 
+    geom_point(data = objective2, aes(x = iteration, y = z), colour = "red") + 
+    geom_tallrect(data = objective2, aes(xmin = iteration - 1 / 2, xmax = iteration + 1 / 2, 
+                                         clickSelects = iteration2), alpha = .3) + 
+    geom_text(data = objective2, aes(x = iteration, y = z + 0.3, showSelected = iteration2, 
+                                     label = iteration, hjust = hjust)) + 
+    ggtitle("objective value vs. iteration") + 
+    theme_animint(width = 600, height = 600)
+  
+  viz <- list(contour = contour.plot, objective = objective.plot, 
+              time = list(variable = "iteration2", ms = 2000), 
+              title = "Demonstration of Gradient Descent Algorithm")
+  info <- animint2HTML(viz)
+}
+
+getStyleValue <- function(html, xpath, style.name) {
+  nodes <- getNodeSet(html, xpath)
+  node.style <- xmlAttrs(nodes[[1]])["style"]
+  pattern <-paste0("(?<name>\\S+?)", ": *", "(?<value>.+?)", ";")
+  style.matrices <- str_match_all_perl(node.style, pattern)
+  style.value <- style.matrices[[1]][style.name, "value"]
+}
+
+test_that('geom_text(hjust=0) => <text style="text-anchor: start">', {
+  info <- grad.desc.viz(hjust = 0)
```

Comment:
in tests it is best to be as explicit as possible, so I would recommend that grad.desc.viz returns an animint list of ggplots and options, and then you take animint2HTML outside of the grad.desc.viz function, e.g.

```r
viz <- grad.desc.viz(hjust=0)
info <- animint2HTML(viz)
```